spec/calendar.html

```diff
@@ -476,7 +518,7 @@ <h1>ToTemporalCalendar ( _temporalCalendarLike_ )</h1>
         1. Let _identifier_ be ? ToString(_temporalCalendarLike_).
         1. Set _identifier_ to ? ParseTemporalCalendarString(_identifier_).
```

Comment:
Spec newbie question: what does it mean to have a `?` before calling an AO, what does it mean when there's a `!`, and why sometimes is there no prefix symbol at all like the call to IsBuiltinCalendar below?

Reply:
> Spec newbie question: what does it mean to have a `?` before calling an AO, what does it mean when there's a `!`, and why sometimes is there no prefix symbol at all like the call to IsBuiltinCalendar below?

`?` and `!` are [ReturnIfAbrupt Shorthands](https://tc39.es/ecma262/multipage/notational-conventions.html#sec-returnifabrupt-shorthands), `?` acting to propagate exceptions from the invocation and `!` asserting the absence of any such exception. They only apply to operations that return Completion Records (which can represent both success and abrupt exception results), and must not appear in front of operations that always return other kinds of results (such as IsBuiltinCalendar, which always returns a **true** or **false** language value).

Reply:
tl;dr,
- `?` = can throw here
- `!` = this operation can throw, but it's not possible to throw here in particular
- nothing = this operation can't throw

Reply:
> `?` and `!` are [ReturnIfAbrupt Shorthands](https://tc39.es/ecma262/multipage/notational-conventions.html#sec-returnifabrupt-shorthands), `?` acting to propagate exceptions from the invocation and `!` asserting the absence of any such exception

Thanks for the explanations! Before commenting above, I read that section of the 262 spec and ended up just as confused as before. :-) 

If I have a spare hour, I'm considering sending an editorial PR over to 262 with a slightly-more-formal variant of @ptomato's explanation, added as a note in the [ReturnIfAbrupt Shorthands](https://tc39.es/ecma262/multipage/notational-conventions.html#sec-returnifabrupt-shorthands) section. I think it could really help other spec newbies to understand the conventions of the spec.

Thanks again.

Reply:
BTW, one reason I was confused by those prefixes is because I think I found a spec bug where `!` shouldn't be there. https://github.com/tc39/proposal-temporal/pull/2515